test/images/serve_hostname/Makefile

```diff
@@ -12,17 +12,45 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Cross-build the serve_hostname image
+#
+# Usage:
+#       [TAG=v1.4] [PREFIX=gcr.io/google_containers] [ARCH=amd64] make container
+
 all: serve_hostname
 
-TAG = v1.4
-PREFIX = gcr.io/google_containers
-TEST_PREFIX = b.gcr.io/k8s_authenticated_test
+TAG?=v1.4
+ARCH?=amd64
+PREFIX?=gcr.io/google_containers
+TEST_PREFIX=b.gcr.io/k8s_authenticated_test
+GOARM=6
+TEMP_DIR:=$(shell mktemp -d)
+KUBE_CROSS_TAG=$(shell cat ../../../build/build-image/cross/VERSION)
+
+ifeq ($(ARCH),amd64)
+        CC=x86_64-linux-gnu-gcc
+endif
+ifeq ($(ARCH),arm)
+        CC=arm-linux-gnueabi-gcc
+endif
+ifeq ($(ARCH),arm64)
+        CC=aarch64-linux-gnu-gcc
+endif
+ifeq ($(ARCH),ppc64le)
+        CC=powerpc64le-linux-gnu-gcc
+endif
 
 serve_hostname: serve_hostname.go
-	CGO_ENABLED=0 go build -a -installsuffix cgo --ldflags '-w' ./serve_hostname.go
+	# Copy the content in this dir to the temp dir
+	cp ./* $(TEMP_DIR)
+
+	docker run -it -v $(TEMP_DIR):/serve_hostname gcr.io/google_containers/kube-cross:$(KUBE_CROSS_TAG) /bin/bash -c \
+	    "cd /serve_hostname \
+	    && CGO_ENABLED=0 GOARM=$(GOARM) GOARCH=$(ARCH) CC=$(CC) CGO_ENABLED=0 go build -a -installsuffix cgo --ldflags '-w' /serve_hostname/serve_hostname.go"
+
 
 container: serve_hostname
-	docker build -t $(PREFIX)/serve_hostname:$(TAG) .
+	docker build -t $(PREFIX)/serve_hostname:$(TAG) $(TEMP_DIR)
```

Comment:
This does still only build for one arch.
It should be `$(PREFIX)/serve_hostname-$(ARCH):$(TAG)`